### ISSUE TITLE
minor changes vs editorial changes

### ISSUE THREAD

**Issue by frivoal** [w3c/w3process#243]
This is essentially a follow up to #111. There are a few  places where the undefined “minor changes“ is used, and where we'd be better off using “editorial changes“, which is actually defined:

* [Section 5.2.3](https://w3c.github.io/w3process/#CharterReview)
  > The Director is not required to solicit Advisory Committee review prior to a charter extension or for minor changes

  -> just use “editorial changes“

* [Section 6.1.1.](https://w3c.github.io/w3process/#recs-and-notes)
  “minor changes“ appears in the diagram but the Process's prose uses “editorial changes“, so we should adjust.

* [Section 6.1.2](https://w3c.github.io/w3process/#maturity-levels)
  > An Edited Recommendation: A Working Group may make editorial **or other minor** changes to a

  We resolved in #111 to only say "editorial changes", but the pull request (#125) that claimed to apply that resolution didn't do that change correctly.

* [Section 6.7](https://w3c.github.io/w3process/#rec-modify)
“minor changes“ appears in the diagram but the Process's prose uses “editorial changes“, so we should adjust.

* [Section 7.1.2](https://w3c.github.io/w3process/#ACReviewAfter)
  > 1. The proposal is approved, possibly with minor changes integrated. 

  This is in contrast with point 2, which talks about substantive changes, which are defined in contrast of editorial changes, so that's the term we should be using.

**Comment by frivoal** [comment]
oops, clicked submit too fast. Adding the rest now...

**Comment by frivoal** [comment]
* [Section 6.1.2](https://w3c.github.io/w3process/#maturity-levels)
  > An Edited Recommendation: A Working Group may make editorial **or other minor** changes to a

  We resolved in #111 to only say "editorial changes", but the pull request (#125) that claimed to apply that resolution didn't do that change correctly.

* [Section 6.7](https://w3c.github.io/w3process/#rec-modify)
“minor changes“ appears in the diagram but the Process's prose uses “editorial changes“, so we should adjust.

* [Section 7.1.2](https://w3c.github.io/w3process/#ACReviewAfter)
  > 1. The proposal is approved, possibly with minor changes integrated. 

  This is in contrast with point 2, which talks about substantive changes, which are defined in contrast of editorial changes, so that's the term we should be using.

**Comment by frivoal** [comment]
Is any of these controversial? If not, I'll do a Pull Request fixing them all.

**Comment by wseltzer** [comment]
-1 to the first of these. 
5.2.3 "minor changes" to a charter could include change of team contact time, acknowledging resignation or affiliation change of a chair. Those non-editorial changes should be reflected in the charter document without requiring AC review.

**Comment by frivoal** [comment]
@wseltzer We could define the things you just listed to be part of what editorial changes are for charters, or create a subclass of substantive changes that would be defined to be minor, if there is a need to distinguish (I am not convinced there is such a need).

What I am uncomfortable with is that "minor changes" are wholly undefined, which effectively grants unlimited authority to change anything in a charter without soliciting the AC, with no defined recourse should someone disagree that a particular change is minor.

**Comment by michaelchampion** [comment]
I'm not opposed to defining "minor changes" but at some point we have to trust the team to do the right thing on Process implementation details. I think the current procedure of circling back to all who voted on a charter to give them an opportunity to object to whatever changes they made to a draft charter in response to comments is sufficient to flag real problems.

**Comment by frivoal** [comment]
Please don't frame this as a matter of trust. I trust the Team to the right thing. That doesn't mean we don't need a Process.

> I think the current procedure of circling back to all who voted on a charter to give them an opportunity to object to whatever changes they made to a draft charter in response to comments is sufficient to flag real problems.

Yes, and we agreed in the AB that that was the right thing to do. However, this bit in 5.2.3 is about a different part of the process that specifically allows the Director to change things in the already approved charter of an active WG without asking or telling anyone, and uses an undefined term to characterize the type of things that can be thus changed.

We already define "editorial changes". "editorial changes" and "minor changes" look similar enough that people can mistake them for synonyms. If we mean them as synonyms, we should use a single term. If we use different terms because we want a difference of meaning, we should characterize what that difference is.

I cannot think of any situation where it is useful to distinguish, so I propose unifying the two terms, possibly after having adjusted the definition to make sure we don't exclude things we intend to include (such as what Wendy proposed).

**Comment by wseltzer** [comment]
For charters specifically, I'd suggest that what we're looking for is "changes that do not affect the scope of work, exit criteria, or decision policy."

**Comment by frivoal** [comment]
Thanks @wseltzer, I think we're getting close. Looking at a few actual charters, I think I'd make the list of things that aren't minor/editorial changes a bit longer, without straying too far from what you said:

> changes that do not affect the scope of work or deliverables; success or exit criteria; communication, work mode, or decision policy; dependencies or liaisons, or Intellectual Property policy."

Otherwise, we could switch a WG to the use the GPL license for specs, not use the patent policy, no longer have physical meetings or use github but have daily 3h teleconferences with member-only minutes, be considered successful without the participation of implementors, say that all the documents worked on will be NOTES rather than RECs, remove the requirement for horizontal review or coordination with any other group, and still call that a minor change :-)

The definition above still leaves us free to change information about the team contact or chairs, the "about this charter" / change section, introductory prose or notes, and various kinds of meta-data in the header or footer. As for the end-date, the Process already classifies it as a substantive change that the Director has the power to extend it unilaterally anyway.

I think this definition is still succinct, and shouldn't get in the way of the Team doing their job. Does that sound reasonable, or do you think I'm excluding some things that should be possible to change unannounced in the middle of the life of a WG? If you do, concrete examples would help me understand.

**Comment by wseltzer** [comment]
I think that's too specific. Here are some counterexamples that I think shouldn't require AC review:

* dependencies and liaisons: a WG with which the chartered WG interacts changes name or splits; a different group takes responsibility for a deliverable this WG was tracking
* communication/work mode: a group has in fact dropped teleconfs and wants to reflect that in the charter document

Really, I think we need to be more just-in-time in our detail. If something becomes problematic, let's address it then.

**Comment by frivoal** [comment]
> I think that's too specific. Here are some counterexamples that I think shouldn't require AC review:
> * dependencies and liaisons: a WG with which the chartered WG interacts changes name or splits; a different group takes responsibility for a deliverable this WG was tracking

Fair enough, I'm OK with dropping that part. In practice, as long we don't bury normative text in there (see https://github.com/w3c/charter-drafts/issues/192), I find that part of charters to be mostly about setting expectations rather than a hard constraint on how the group works, so low overhead in keeping it updated outweighs the risk of a controversial change.

> * communication/work mode: a group has in fact dropped teleconfs and wants to reflect that in the charter document

Here I am less convinced. Different people and different communities work differently, for reasons ranging from culture, to personal preferences, to accessibility. Changing the work mode may also change the necessary time commitments, or which geography finds it easy or hard to participate. If a group was set up with "may hold teleconferences", it is up to them to have them or not. But if the charter say that they must, or that they must not, presumably that was done for a reason. Charters should not over-constrain the way groups work for no good reason, but when they place some requirements or constraints, these should be upheld.

Groups should be allowed to evolve, but these parts are material to the ability of a group conducts its work. Letting groups change how they do things in contradiction with their charter, and the Team/Director change the charter accordingly without input from those who voted the charter into being seems to defeat the point of having it in the charter in the first place.

If the AB/ProcessCG disagrees with me on this point, I'll yield, but the very fact that it's not obvious is why I want it to be defined one way or another.

> Really, I think we need to be more just-in-time in our detail. If something becomes problematic, let's address it then.

Well, this part is about allowing the Director to bypass the normal charter updating process and change things with low overhead by neither having to ask or tell anyone. While I think it is reasonable for this mechanism to exist in order to get simple things done without undue burden, I don't think it is good to allow that without characterizing what sort of things may be thus changed.

The context for this kind of clean-up and disambiguation is as a prelude to working on the role of the director. This is one of a few places where the process boils down to letting the Director do anything at all without control or limitation, which is OK because the founding Director's unique role. But as we prepare to consider how the powers of the Director may be distributed, it helps to know what these powers actually are.

----

With that in mind, dropping dependencies and liaisons, but keeping communication/work-mode, here's what we get:

> changes that do not affect the scope of work or deliverables; success or exit criteria; communication, work mode, or decision policy; or Intellectual Property policy.

**Comment by dwsinger** [comment]
Perhaps we can say that the changes that the Director makes without review can be operational, editorial, or minor, but not substantive, and that (for the avoidance of doubt) " the scope of work (and hence the list of deliverables), exit criteria, decision policy, choice of license or nature of deliverable, are all substantive"?  (e.g. a change from a WG note to Rec track would be substantive, or a change from the W3C document license to CC-BY, or...)

**Comment by chaals** [comment]
I share @frivoal 's concern about changing work modes established by charter. In any event, I don't think the Director or Team should be unilaterally directing any such changes, so they should be reviewed.

The changes to charters are not editorial. I think we should use something like @wseltzer's or @dwsinger's formulation to identify them differently.

**Comment by wseltzer** [comment]
@dwsinger as a side note, the scope of work is not coextensive with the list of deliverables. Without changing charter, groups can already add new deliverables that fit within their defined scope, unless the charter explicitly prohibits it.

**Comment by dwsinger** [comment]
you're right, the charter can allow splitting of work and publication of other deliverables, as long as the charter allows them. 

"expansion of the scope of work, changes to exit criteria, group decision policy, choice of license, addition of normative deliverables not in the scope of the current charter, are substantive"…

(examples are dangerous)

**Comment by frivoal** [comment]
That sort of changes to the deliverables is fine. What wouldn't be fine is if the charter prohibited adding new deliverables, and the Team/Director removed that restriction without asking, or if it wasn't and the Team/Director added it.

@dwsinger I don't think there's any situation where we'd editorial changes differently from minor changes. So, we we could distinguish between editorial vs minor vs substantive, and in the prose that uses these terms, always say `editorial and minor changes`, or we could define `editorial` to cover all the cases we care about, and only use that term. I have a preference for the latter, as I don't see the need to introduce separate terminology if we're not going to do anything with it, but it's just a preference, I wouldn't oppose the alternative.

As for what we define the editorial / minor changes to be, I earlier suggested:

> changes that do not affect the scope of work or deliverables; success or exit criteria; communication, work mode, or decision policy; or Intellectual Property policy.

That seems to be substantially the same as your list, but I have a slight preference for my own phrasing:
* "Intellectual Property" is intentionally broader than "choice of license", cause we don't want to say that changes to the Patent Policy, or adding a trademark licensing requirement would be editorial/minor.
* I do include deliverables. If the charter itself allows for deliverables to evolve, then evolving the deliverables is not a change in charter. If it doesn't then it's not a minor/editorial change.
* In addition to "exit criteria", which are about specs, I've included "success criteria", which are about the group itself
* I've listed "communication, work mode, or decision policy" rather than just "group decision policy". To some degree there's overlap, but switching from public to private proceedings, or changing from having meetings to not having them, or vice versa, isn't minor/editorial, so I'd rather include it.

**Comment by frivoal** [comment]
Agenda+ to try and answer two questions:
* Do we go with Dsinger's phrasing, or mine, or some hybrid
* Do we define a separate "minor changes" term, or do we fold these into "editorial"

**Comment by chaals** [comment]
I like @frivoal 's phrasing, and I agree with each of the arguments made in [the comment above](https://github.com/w3c/w3process/issues/243#issuecomment-472313492). I have a strong preference for "minor" being defined and used, because I think "editorial" changes have a fairly clear meaning even in plain english and they don't cover things we might allow for.

I wonder if significant changes in team contact time should be "minor". My experience is that often the numbers in the charter are fiction - in which case they should not be framed the way they are - but where there is a significant commitment changes to that have a significant impact on the WG. It should be relatively easy to make such changes, but they should be explained to the AC ("the Team member found a better job", "the group is much less dependent than before/anticipated/we want", etc). I don't **think** that question needs to hold up this issue.

I also wonder if we should frame "minor changes" differently, pointing out that they are not meant to affect the way the group functions in any significant way, and thus exclude [the list of things] **or similarly impactful changes**. I'm trying to avoid the situation where a change is claimed to be minor just because it isn't on the list, e.g. "Web Platform WG is not going to publish any of its deliverables for at leat six months" - which should be in charter extensions and should not be considered minor.

**Comment by dwsinger** [comment]
I am lost on what the proposed fix is, as a result of these discussions. I can't tell any longer; there are some accepted comments related to @frivoal 's original phrasing, and concerns and support about mine, so...where are we?

**Comment by frivoal** [comment]
Where I'm at is that I think your phrasing would be an improvement over the undefined status quo, but I prefer my phrasing of the list, and chaals agrees. I originally envisioned bundling that into the second type of editorial changes in [section 6.2.5. “Classes of Changes”](https://w3c.github.io/w3process/#correction-classes), but I am OK with incorporating chaal's suggestion that it should rather be described as minor changes separately from editorial changes, as well as with his proposed addition of "or similarly impactful changes" phrasing.

Which in full, gives:
> Non-editorial changes to a Charter are considered minor when they do not affect the way the group functions in any significant way, and thus exclude changes to scope of work or deliverables; success or exit criteria; communication, work mode, or decision policy; Intellectual Property policy; or similarly impactful changes.

**Comment by dwsinger** [comment]
OK, as Wendy points out, splitting a deliverable (and hence adding one) or adding a deliverable that's clearly in the original scope are probably minor, no? Maybe it would be safer also to insert "for example" to emphasize that the list does not purport to be 100% complete or exact?

Non-editorial changes to a Charter are considered minor when they do not affect the way the group functions in any significant way, and thus, for example, exclude changes to scope of work; addition of normative deliverables not in the scope of the current charter; changes to success or exit criteria; changes to communication, work mode, or decision policy; changes to Intellectual Property policy; or similarly impactful changes.

**Comment by nigelmegitt** [comment]
@frivoal the wording in https://github.com/w3c/w3process/issues/243#issuecomment-473145152 seems grammatically the wrong way around to me.

It says that minor non-editorial changes exclude [list of scenarios]. I think it needs to say "[list of scenarios] are therefore not considered minor".

So:

> Non-editorial changes to a Charter are considered minor when they do not affect the way the group functions in any significant way. Therefore the following are examples of non-minor non-editorial changes: changes to scope of work or deliverables; success or exit criteria; communication, work mode, or decision policy; Intellectual Property policy; or similarly impactful changes.

**Comment by frivoal** [comment]
> splitting a deliverable (and hence adding one) or adding a deliverable that's clearly in the original scope are probably minor, no?

I think it depends how you understand the word "scope" :)

If a charter says that the group can add deliverables as it goes along, then doing so does not need a change of charter (and the list should probably be maintained somewhere else than the charter). If a charter says that the group will work on these 3 things and these only, then the addition of one is not a minor change, even if the new one would fall under the conceptual scope of the WG. See for the example the discussions we've had about the SVG-WG, where various people have insisted at various points that SVG2 should be the only deliverable of the group.

Maybe you're right to characterize that in terms of scope, but at the same time I think charters often don't frame it this way, and use scope in a more abstract manner.

With this clarification about my intent, do you still think your wording covers it, or should we try and find some variation?

**Comment by dwsinger** [comment]
I think someone has to exercise good judgment. Imagine "the Fruits and Vegetables specification was getting too big, and we realized that we could reasonably split into two documents; climeractic fruits in one, and non-climeractic fruits and vegs in the other". No change of scope, simply two docs where there was one. Or "The WG will develop a set of specifications documenting the application of any document AES encryption mode to the secure exchange mechanism; the initial set will cover CBC and CFB but documents for other modes will be added if there is both demand and expertise." So, an announcement that they are adding a document on CTR modes is no surprise to anyone and shouldn't cause a ballot; the scope clearly said all AES modes were in scope.

**Comment by frivoal** [comment]
Given the addition of “similarly impactful changes” and “for example“, I think you're right, and that we can just not talk about deliverables. Let's go ahead with that?

On a minor note, which phrasing do we prefer? That of @dwsinger in https://github.com/w3c/w3process/issues/243#issuecomment-473346362, or that of @nigelmegitt in https://github.com/w3c/w3process/issues/243#issuecomment-473453564 (with "or deliverables" removed)?

I have no strong preference, and since it's a grammar tweak, I'd rather leave it up to native speakers.

**Comment by dwsinger** [comment]
Merging Nigel's better grammar with my expanded list:

Non-editorial changes to a Charter are considered minor when they do not affect the way the group functions in any significant way. Therefore the following are examples of non-minor non-editorial changes: changes to the scope of work; addition of normative deliverables not in the scope of the current charter; changes to success or exit criteria; changes to communication, work mode, or decision policy; changes to Intellectual Property policy; or similarly impactful changes.

**Comment by wseltzer** [comment]
Too many double negatives. 
"None of the following therefore qualifies as a minor editorial change:..." 

(Note that I still disagree with the update overall, but at least let's make it clearer to read.)

**Comment by fantasai** [comment]
Merging in with the proposed text:

> Non-editorial changes to a Charter are considered minor when they do not affect the way the group functions in any significant way. Therefore the following are <em>not</em> examples of minor editorial changes: changes to the scope of work; addition of normative deliverables not in the scope of the current charter; changes to success or exit criteria; changes to communication, work mode, or decision policy; changes to Intellectual Property policy; or similarly impactful changes.

**Comment by dwsinger** [comment]
Non-editorial changes to a Charter are considered minor when they do not affect the way the group functions in any significant way. Therefore the following are examples of changes that are not minor nor editorial: changes to the scope of work; addition of normative deliverables not in the scope of the current charter; changes to success or exit criteria; changes to communication, work mode, or decision policy; changes to Intellectual Property policy; or similarly impactful changes.

**Comment by dwsinger** [comment]
tentatively resolved in May 2019 call, but further discussion with Wendy is suggested

**Comment by dwsinger** [comment]
um, "not (examples of X)" and "examples of (not X)" are not equivalent. Somehow Elika's comment managed to get into the future, but I think the last word is https://github.com/w3c/w3process/issues/243#issuecomment-490515377

**Comment by nigelmegitt** [comment]
Not sure if there's a style guide for this, but I would switch "not minor nor editorial" to "neither minor nor editorial" for grammar.

**Comment by dwsinger** [comment]
Non-editorial changes to a Charter are considered minor when they do not affect the way the group functions in any significant way. Therefore the following are examples of changes that are neither minor nor editorial: changes to the scope of work; addition of normative deliverables not in the scope of the current charter; changes to success or exit criteria; changes to communication, work mode, or decision policy; changes to Intellectual Property policy; or similarly impactful changes.

**Comment by frivoal** [comment]
I've talked with Wendy, and made a pull request based on that discussion that is OK with her: #269. It's not identical to the proposal arrived to in this thread, but it's not in contradiction to it either.

I suggest reading the 4 commits one by one, as their commit messages tries to explain what I'm doing.

Broadly speaking though, the approach taken is based on the realization that there really is only one place in the entire Process where the "minor changes" distinction serves a useful purpose: [5.2.3 ](https://w3c.github.io/w3process/#CharterReview), with every other being either pointless (as in 7.1.2), or mistakes that should refer to editorial changes instead.

So, this PR gets rid of the rest, and since we're left with a single use, we can say in place what we mean, without crafting a generic term that might get used in different contexts.

As for what we mean in that particular place, I'm reusing the normative phrasing we came to in this issue ("do not affect the way the group functions in any significant way"), but for examples, instead of the negative list of changes that wouldn't be minor, Wendy recommended that we go for a positive list of changes that would be, as that better highlights what this provision is intended for. Also, my understanding is that she felt that a negative list would call for more precision to avoid accidentally banning otherwise minor tweaks to important areas.

With the scope being reduced by inlining this definition in the one place where in matters, I believe that @wseltzer could agree to the phrasing in [the last comment by @dwsinger](https://github.com/w3c/w3process/issues/243#issuecomment-490977887) with the negative examples, but that she would prefer if we could use the one I've put in the PR with the positive ones.

@wseltzer, please confirm I have not misrepresented your position.

Everyone else, feedback appreciated.

**Comment by dwsinger** [comment]
I'm OK with this approach, and I like not using "minor changes" at all.

**Comment by nigelmegitt** [comment]
I like the overall approach too.

The sense of the proposal in https://github.com/w3c/w3process/issues/243#issuecomment-490977887 seems to have been lost a little - whereas some non-minor non-editorial changes were explicitly listed there, the result of the proposed change is only to list examples of minor changes, which is a much smaller list. 

This means that if there was ambiguity about whether, say, a change to the decision policy, needs an AC review, that ambiguity has not been resolved by #269 as it currently stands. This seems like a shame.

**Comment by frivoal** [comment]
@wseltzer, I don't want to put words into your mouth. Can you respond to @nigelmegitt above and explain why you prefer listing examples of acceptable changes to listing examples of non acceptable ones?

**Comment by wseltzer** [comment]
@nigelmegitt I feared that the list would be read as limiting (even if the language around it tried to be non-exhaustive). I also don't want to proceduralize questions that haven't yet arisen -- in other words,  ambiguity is the right response to an abstract question to which either answer seems plausible.

**Comment by nigelmegitt** [comment]
@wseltzer thanks for the explanation. I must admit I had _assumed_ that the examples were present because questions about them had indeed arisen.

**Comment by frivoal** [comment]
@dwsinger (with chair hat on), and other participants, I'd like to know if this is ready to merge or not. The PR is in line with what @wseltzer wanted, and differs  from the text we were converging on only in the way examples are framed. Are we good to go, or do we want to further discuss which way we want the examples to be written?

(personally, I prefer the original phrasing, but I'm also OK with the one preferred by @wseltzer, so I'd like to merge and move on.)
